blocks/editable/format-toolbar/index.js

```diff
@@ -42,6 +42,13 @@ const FORMATTING_CONTROLS = [
 // Default controls shown if no `enabledControls` prop provided
 const DEFAULT_CONTROLS = [ 'bold', 'italic', 'strikethrough', 'link' ];
 
+/* We need to stop the keypress event here, because block.js is firing
+	 a maybeStartTyping on keypress, and that hides the "fixed-to-block" toolbar
+	 which unregisters the slot, so when Editable tries to re-render its input
+	 dialog, the slot is no longer in the system, and the dialog disappears
+ */
```

Comment:
nit: Could you shorten this to a one line comment? It's enough that the comment gives the reader a quick heads up so that they can e.g. run a `git blame` if they'd like to learn more about the bug fix.

My suggestion:

```js
// Stop the keypress event from propagating up to maybeStartTyping in BlockListBlock
const stopKeyPressPropagation = ( event ) => event.stopPropagation();
```